GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -226,6 +225,27 @@ string ComposeOutput()
             }
         }
 
+        /// <summary>
+        /// Launches a process for the executable and returns output lines as they become available.
+        /// </summary>
+        /// <param name="executable">The executable from which to launch a process.</param>
+        /// <param name="arguments">The arguments to pass to the executable</param>
+        /// <param name="writeInput">A callback that writes bytes to the process's standard input stream, or <c>null</c> if no input is required.</param>
+        /// <param name="outputEncoding">The text encoding to use when decoding bytes read from the process's standard output and standard error streams, or <c>null</c> if the default encoding is to be used.</param>
+        /// <param name="stripAnsiEscapeCodes">A flag indicating whether ANSI escape codes should be removed from output strings.</param>
+        /// <returns>An enumerable sequence of lines that yields lines as they become available. Lines from standard output are returned first, followed by lines from standard error.</returns>
+        [MustUseReturnValue("If output lines are not required, use " + nameof(RunCommand) + " instead")]
+        public static async Task<IEnumerable<string>> GetOutputLinesAsync(
+            this IExecutable executable,
+            ArgumentString arguments = default,
+            Action<StreamWriter> writeInput = null,
+            Encoding outputEncoding = null,
+            bool stripAnsiEscapeCodes = true)
+        {
+            var result = await executable.ExecuteAsync(arguments, writeInput, outputEncoding, stripAnsiEscapeCodes);
```

Comment:
It is not an async stream, but at least is non blocking.

Reply:
What kind of stream is it? What would it take to make it properly asynchronous?

Reply:
I believe it is https://docs.microsoft.com/pl-pl/dotnet/api/system.collections.generic.iasyncenumerable-1?view=netcore-3.0